x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -492,6 +492,39 @@ export class KnowledgeBaseService {
     };
   };
 
+  getInstructions = async (): Promise<UserInstruction[]> => {
```

Comment:
can we leverage the existing `recall` method on the knowledge base service? that also includes RBAC

Reply:
I can add the RBAC filter to this query as well.

I chose to add a new method here so that instructions are available **before** the first recall, in case users may want to avoid using recall or even tweak the actions done in recall (since recall uses the same system message).

Adding it to recall makes it directly available in telemetry and copy conversation **but** makes it so that I have check if the last message in the `next` loop is the response from recall, grab the instructions and then extend the system message before calling `next` again, which means also repeating this step for each time we do recall. 

I felt doing it this way was simpler but I can make the changes if you prefer.

Reply:
Based on talking in Slack, since `recall` also hits `search-*` we decided to leave it this way **but** add in the same RBAC control.

Reply:
Fixed in https://github.com/elastic/kibana/pull/180263/commits/572f751e9317dc327eeb9e3ec7b554fe2f4ca00d